### PR TITLE
Adding a Vagrantfile to start a development VM

### LINKED ISSUE #4: Creating a vagrantfile
We need a unified development environment for everyone
working on the project, therefore, using vagrant we will all
use the same type of machine and achieve this.
Required outcome: everyone should be able to clone the repo,
'vagrant up', and start the VM.

### PR DESCRIPTION
**Pull request by Davidco94**
1. Adding a Vagrantfile to spin up a fedora-34 development machine.
2. Making changes in the '.gitignore' file to ignore internal files and data that Vagrant1 creates.
fixes #4

### CODE REVIEW

**Review by idan1356 — APPROVED** [review]
it appears the vagrant file and git ignore addition were configured properly, in accordance with the tool introduction challenge
fix #4

**Comment by Davidco94** [comment]
We need a unified development environment for everyone
working on the project, therefore, using vagrant we will all
use the same type of machine and achieve this.
Required outcome: everyone should be able to clone the repo,
'vagrant up', and start the VM.

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+# backwards compatibility). Please don't change it unless you know what
+# you're doing.
+Vagrant.configure("2") do |config|
+  # The most common configuration options are documented and commented below.
+  # For a complete reference, please see the online documentation at
+  # https://docs.vagrantup.com.
+
+  # Every Vagrant development environment requires a box. You can search for
+  # boxes at https://vagrantcloud.com/search.
+  config.vm.box = "fedora/34-cloud-base"
```
Comment (natifridman) [inline Vagrantfile:15]: This is fedora34 and not 32 as mentioned in the PR description and your commit message

File: Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  # config.vm.box_download_insecure = true
+  config.vm.box_version = "34.20210423.0"
```
Comment (natifridman) [inline Vagrantfile:17]: Why do you need to use a specific version? what's wrong with the default?

**Review by natifridman — COMMENTED** [review]
Nice work with a detailed description here and in the commit message.
Added some comments regarding the VM version

File: Vagrantfile (hunk 1 above)
Comment (Davidco94) [reply Vagrantfile:15]: I fixed it

Comment (natifridman) [reply Vagrantfile:15]: Could you please change it in the commit message also? You can use [this guide](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) for rewriting git history?

File: Vagrantfile (hunk 2 above)
Comment (natifridman) [reply Vagrantfile:17]: Ok, so let's keep it as it is

File: Vagrantfile (hunk 1 above)
Comment (Davidco94) [reply Vagrantfile:15]: I did it..
I hope what I did was right

**Review by natifridman — APPROVED** [review]
LGTM

File: Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,78 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+# All Vagrant configuration is done below. The "2" in Vagrant.configure
```
Comment (Omeramsc) [inline Vagrantfile:4]: please move all the template comments in the Vargrantfile

**Pull request merged**